Source/Scene/Instanced3DModel3DTileContent.js

```diff
@@ -189,109 +202,217 @@ define([
-        byteOffset += gltfByteLength;
-
-        // Each vertex has a longitude, latitude, and optionally batchId if there is a batch table
-        // Coordinates are in double precision, batchId is a short
-        var instanceByteLength = sizeOfFloat64 * 2 + (hasBatchTable ? sizeOfUint16 : 0);
-        var instancesByteLength = instancesLength * instanceByteLength;
-
-        var instancesView = new DataView(arrayBuffer, byteOffset, instancesByteLength);
-        byteOffset += instancesByteLength;
-
-        // Create model instance collection
-        var collectionOptions = {
-            instances : new Array(instancesLength),
-            batchTableResources : batchTableResources,
-            boundingVolume : this._tile.contentBoundingVolume.boundingVolume,
-            cull : false,
-            url : undefined,
-            headers : undefined,
-            type : RequestType.TILES3D,
-            gltf : undefined,
-            basePath : undefined
-        };
-
-        if (gltfFormat === 0) {
-            var gltfUrl = getStringFromTypedArray(gltfView);
-            collectionOptions.url = joinUrls(this._tileset.baseUrl, gltfUrl);
-        } else {
-            collectionOptions.gltf = gltfView;
-            collectionOptions.basePath = this._url;
-        }
+        if (featureTableJSONByteLength > 0) {
+            var featureTableString = getStringFromTypedArray(uint8Array, byteOffset, featureTableJSONByteLength);
+            var featureTableJSON = JSON.parse(featureTableString);
+            byteOffset += featureTableJSONByteLength;
 
-        var ellipsoid = Ellipsoid.WGS84;
-        var position = new Cartesian3();
-        var instances = collectionOptions.instances;
-        byteOffset = 0;
-
-        for (var i = 0; i < instancesLength; ++i) {
-            // Get longitude and latitude
-            var longitude = instancesView.getFloat64(byteOffset, true);
-            byteOffset += sizeOfFloat64;
-            var latitude = instancesView.getFloat64(byteOffset, true);
-            byteOffset += sizeOfFloat64;
-            var height = 0.0;
-
-            // Get batch id. If there is no batch table, the batch id is the array index.
-            var batchId = i;
-            if (hasBatchTable) {
-                batchId = instancesView.getUint16(byteOffset, true);
-                byteOffset += sizeOfUint16;
-            }
+            var featureTableBinary = new Uint8Array(arrayBuffer, byteOffset, featureTableBinaryByteLength);
+            byteOffset += featureTableBinaryByteLength;
 
-            Cartesian3.fromRadians(longitude, latitude, height, ellipsoid, position);
-            var modelMatrix = Transforms.eastNorthUpToFixedFrame(position);
+            var featureTableResources = new Cesium3DTileFeatureTableResources(featureTableJSON, featureTableBinary);
+            var instancesLength = featureTableResources.getGlobalProperty('INSTANCES_LENGTH', WebGLConstants.UNSIGNED_INT);
```

Comment:
`ComponentDatatype` might be clearer to use. As a side note, we should definitely add `INT` and `UNSIGNED_INT` to `ComponentDatatype` so that we can actually load integer properties since the feature table calls `ComponentDatatype.createArrayBufferView`.